src/main/groovy/com/athaydes/spockframework/report/vivid/VividAstInspector.groovy

```diff
@@ -0,0 +1,250 @@
+package com.athaydes.spockframework.report.vivid
+
+import org.codehaus.groovy.ast.ASTNode
+import org.codehaus.groovy.ast.AnnotatedNode
+import org.codehaus.groovy.ast.AnnotationNode
+import org.codehaus.groovy.ast.ClassCodeVisitorSupport
+import org.codehaus.groovy.ast.ClassNode
+import org.codehaus.groovy.ast.ConstructorNode
+import org.codehaus.groovy.ast.FieldNode
+import org.codehaus.groovy.ast.MethodNode
+import org.codehaus.groovy.ast.ModuleNode
+import org.codehaus.groovy.ast.Parameter
+import org.codehaus.groovy.ast.PropertyNode
+import org.codehaus.groovy.ast.expr.ArgumentListExpression
+import org.codehaus.groovy.ast.expr.ConstantExpression
+import org.codehaus.groovy.ast.expr.Expression
+import org.codehaus.groovy.ast.expr.MethodCallExpression
+import org.codehaus.groovy.ast.expr.StaticMethodCallExpression
+import org.codehaus.groovy.ast.stmt.ExpressionStatement
+import org.codehaus.groovy.ast.stmt.Statement
+import org.codehaus.groovy.control.CompilationFailedException
+import org.codehaus.groovy.control.CompilationUnit
+import org.codehaus.groovy.control.CompilePhase
+import org.codehaus.groovy.control.CompilerConfiguration
+import org.codehaus.groovy.control.SourceUnit
+import org.spockframework.compiler.AstUtil
+import org.spockframework.compiler.SourceLookup
+import org.spockframework.util.inspector.AstInspector
+import org.spockframework.util.inspector.AstInspectorException
+import org.spockframework.util.inspector.Inspect
+
+import java.security.CodeSource
+
+/**
+ * Based on org.spockframework.util.inspector.AstInspector by Peter Niederwieser
+ */
+public class VividAstInspector extends AstInspector{
```

Comment:
This is codename for this feature. Initially it was "descriptive" for descriptive and meaningful code but vivid is shorter one.